src/crystal/system/win32/visual_studio.cr

```diff
@@ -0,0 +1,51 @@
+require "json"
+
+module Crystal::System::VisualStudio
+  struct Installation
+    include JSON::Serializable
+
+    @[JSON::Field(key: "installationPath")]
+    getter directory : String
+
+    @[JSON::Field(key: "installationVersion")]
+    getter version : String
+
+    # unused fields not mapped
+  end
+
+  def self.find_latest_msvc_path : ::Path?
+    # ported from https://github.com/microsoft/vswhere/wiki/Find-VC
+    # Copyright (C) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+    if vs_installations = get_vs_installations
+      vs_installations.sort_by! &.version.split('.').map(&.to_i)
```

Comment:
What if the versions have non-integers somehow? The compiler will just crash? If you're sure that such versions should never appear then maybe better to skip those, otherwise we'll need to keep thinking about the approach

Reply:
```suggestion
      vs_installations.sort_by! &.version.scan(/[0-9]+/).map(&.[0].to_i)
```

Reply:
The most "correct" approach, according to vswhere: https://github.com/microsoft/vswhere/blob/d38b7b23060a8aa84a9def3730b9a13b6c16f475/src/vswhere.lib/InstanceSelector.cpp#L34-L96